### ISSUE TITLE
Different convergence between using Gluon trainer and symbolic optimizer for training ResNet50V1b

### ISSUE THREAD

**Issue by wuxun-zhang** [dmlc/gluon-cv#1329]
FYI. @pengzhao-intel @TaoLv @xinyu-intel 

Recently, I was training ResNet50-v1b using Horovod+"[sgdwfastlars](https://github.com/mlperf/training_results_v0.6/blob/master/NVIDIA/benchmarks/resnet/implementations/mxnet/common/fit.py#L315)" optimizer. But I found there would be much difference on accuracy between using Gluon Trainer and symbolic trainer.

**Using Gluon Trainer**:
```
[2020-05-27 10:53:20] Epoch [0] Rank [0] Batch [19]     accuracy=0.002344       loss=6.886011   lr=0.063898
[2020-05-27 10:55:18] Epoch [0] Rank [0] Batch [39]     accuracy=0.003320       loss=6.826396   lr=0.127796
[2020-05-27 10:57:14] Epoch [0] Rank [0] Batch [59]     accuracy=0.005599       loss=6.714621   lr=0.191693
[2020-05-27 10:59:12] Epoch [0] Rank [0] Batch [79]     accuracy=0.006836       loss=6.623612   lr=0.255591
[2020-05-27 11:01:08] Epoch [0] Rank [0] Batch [99]     accuracy=0.008125       loss=6.534773   lr=0.319489
[2020-05-27 11:03:05] Epoch [0] Rank [0] Batch [119]    accuracy=0.009049       loss=6.442341   lr=0.383387
[2020-05-27 11:05:02] Epoch [0] Rank [0] Batch [139]    accuracy=0.009821       loss=6.502010   lr=0.447284
[2020-05-27 11:07:00] Epoch [0] Rank [0] Batch [159]    accuracy=0.011133       loss=6.450672   lr=0.511182
[2020-05-27 11:08:57] Epoch [0] Rank [0] Batch [179]    accuracy=0.012370       loss=6.245819   lr=0.575080
[2020-05-27 11:10:55] Epoch [0] Rank [0] Batch [199]    accuracy=0.013555       loss=6.163465   lr=0.638978
[2020-05-27 11:12:52] Epoch [0] Rank [0] Batch [219]    accuracy=0.014702       loss=6.323869   lr=0.702875
[2020-05-27 11:14:49] Epoch [0] Rank [0] Batch [239]    accuracy=0.015853       loss=6.220434   lr=0.766773
[2020-05-27 11:16:46] Epoch [0] Rank [0] Batch [259]    accuracy=0.017188       loss=6.251002   lr=0.830671
[2020-05-27 11:18:43] Epoch [0] Rank [0] Batch [279]    accuracy=0.018527       loss=6.257105   lr=0.894569
[2020-05-27 11:20:40] Epoch [0] Rank [0] Batch [299]    accuracy=0.019740       loss=5.986939   lr=0.958466
[2020-05-27 10:53:20] Epoch [0] Rank [0] Batch [19]     accuracy=0.002344       loss=6.886011   lr=0.063898
[2020-05-27 10:55:18] Epoch [0] Rank [0] Batch [39]     accuracy=0.003320       loss=6.826396   lr=0.127796
[2020-05-27 10:57:14] Epoch [0] Rank [0] Batch [59]     accuracy=0.005599       loss=6.714621   lr=0.191693
[2020-05-27 10:59:12] Epoch [0] Rank [0] Batch [79]     accuracy=0.006836       loss=6.623612   lr=0.255591
[2020-05-27 11:01:08] Epoch [0] Rank [0] Batch [99]     accuracy=0.008125       loss=6.534773   lr=0.319489
[2020-05-27 11:03:05] Epoch [0] Rank [0] Batch [119]    accuracy=0.009049       loss=6.442341   lr=0.383387
[2020-05-27 11:05:02] Epoch [0] Rank [0] Batch [139]    accuracy=0.009821       loss=6.502010   lr=0.447284
[2020-05-27 11:07:00] Epoch [0] Rank [0] Batch [159]    accuracy=0.011133       loss=6.450672   lr=0.511182
[2020-05-27 11:08:57] Epoch [0] Rank [0] Batch [179]    accuracy=0.012370       loss=6.245819   lr=0.575080
[2020-05-27 11:10:55] Epoch [0] Rank [0] Batch [199]    accuracy=0.013555       loss=6.163465   lr=0.638978
[2020-05-27 11:12:52] Epoch [0] Rank [0] Batch [219]    accuracy=0.014702       loss=6.323869   lr=0.702875
[2020-05-27 11:14:49] Epoch [0] Rank [0] Batch [239]    accuracy=0.015853       loss=6.220434   lr=0.766773
[2020-05-27 11:16:46] Epoch [0] Rank [0] Batch [259]    accuracy=0.017188       loss=6.251002   lr=0.830671
[2020-05-27 11:18:43] Epoch [0] Rank [0] Batch [279]    accuracy=0.018527       loss=6.257105   lr=0.894569
[2020-05-27 11:20:40] Epoch [0] Rank [0] Batch [299]    accuracy=0.019740       loss=5.986939   lr=0.958466
[2020-05-27 11:23:08] Rank [0] correct_count = 1988  total_count = 50000
[2020-05-27 11:23:08] Epoch [0] Rank [0] validation: top1-acc=0.039760
[2020-05-27 11:25:05] Epoch [1] Rank [0] Batch [19]     accuracy=0.045703       loss=6.008663   lr=1.063898
[2020-05-27 11:27:02] Epoch [1] Rank [0] Batch [39]     accuracy=0.046680       loss=5.754982   lr=1.127796
[2020-05-27 11:29:00] Epoch [1] Rank [0] Batch [59]     accuracy=0.046875       loss=5.888123   lr=1.191693
[2020-05-27 11:30:57] Epoch [1] Rank [0] Batch [79]     accuracy=0.050098       loss=5.742591   lr=1.255591
[2020-05-27 11:32:54] Epoch [1] Rank [0] Batch [99]     accuracy=0.050937       loss=5.642124   lr=1.319489
[2020-05-27 11:34:51] Epoch [1] Rank [0] Batch [119]    accuracy=0.052865       loss=5.701497   lr=1.383387
[2020-05-27 11:36:48] Epoch [1] Rank [0] Batch [139]    accuracy=0.055134       loss=5.540412   lr=1.447284
[2020-05-27 11:38:45] Epoch [1] Rank [0] Batch [159]    accuracy=0.057959       loss=5.737912   lr=1.511182
[2020-05-27 11:40:41] Epoch [1] Rank [0] Batch [179]    accuracy=0.060547       loss=5.405982   lr=1.575080
[2020-05-27 11:42:38] Epoch [1] Rank [0] Batch [199]    accuracy=0.062578       loss=5.555668   lr=1.638978
[2020-05-27 11:44:36] Epoch [1] Rank [0] Batch [219]    accuracy=0.064205       loss=5.423982   lr=1.702875
[2020-05-27 11:46:33] Epoch [1] Rank [0] Batch [239]    accuracy=0.066764       loss=5.415370   lr=1.766773
[2020-05-27 11:48:30] Epoch [1] Rank [0] Batch [259]    accuracy=0.069832       loss=5.371351   lr=1.830671
[2020-05-27 11:50:27] Epoch [1] Rank [0] Batch [279]    accuracy=0.072126       loss=5.338600   lr=1.894569
[2020-05-27 11:52:24] Epoch [1] Rank [0] Batch [299]    accuracy=0.075234       loss=5.271710   lr=1.958466
[2020-05-27 11:54:52] Epoch [1] Rank [0] validation: top1-acc=0.100400
[2020-05-27 11:56:50] Epoch [2] Rank [0] Batch [19]     accuracy=0.107422       loss=5.260752   lr=2.063898
[2020-05-27 11:58:47] Epoch [2] Rank [0] Batch [39]     accuracy=0.112891       loss=4.950870   lr=2.127796
[2020-05-27 12:00:44] Epoch [2] Rank [0] Batch [59]     accuracy=0.120833       loss=5.326674   lr=2.191693
[2020-05-27 12:02:41] Epoch [2] Rank [0] Batch [79]     accuracy=0.123242       loss=5.114155   lr=2.255591
[2020-05-27 12:04:39] Epoch [2] Rank [0] Batch [99]     accuracy=0.126953       loss=5.273483   lr=2.319489
[2020-05-27 12:06:36] Epoch [2] Rank [0] Batch [119]    accuracy=0.127409       loss=5.235615   lr=2.383387
[2020-05-27 12:08:33] Epoch [2] Rank [0] Batch [139]    accuracy=0.126562       loss=5.255755   lr=2.447284
[2020-05-27 12:10:30] Epoch [2] Rank [0] Batch [159]    accuracy=0.127490       loss=5.067953   lr=2.511182
[2020-05-27 12:12:27] Epoch [2] Rank [0] Batch [179]    accuracy=0.129167       loss=5.137685   lr=2.575080
[2020-05-27 12:14:24] Epoch [2] Rank [0] Batch [199]    accuracy=0.130586       loss=4.955675   lr=2.638978
[2020-05-27 12:16:21] Epoch [2] Rank [0] Batch [219]    accuracy=0.131179       loss=5.047760   lr=2.702875
[2020-05-27 12:18:17] Epoch [2] Rank [0] Batch [239]    accuracy=0.132487       loss=5.035458   lr=2.766773
[2020-05-27 12:20:14] Epoch [2] Rank [0] Batch [259]    accuracy=0.135036       loss=4.976862   lr=2.830671
[2020-05-27 12:22:11] Epoch [2] Rank [0] Batch [279]    accuracy=0.135965       loss=5.214759   lr=2.894569
[2020-05-27 12:24:07] Epoch [2] Rank [0] Batch [299]    accuracy=0.136953       loss=5.170412   lr=2.958466

```

**Using symbolic trainer**
```
2020-05-30 22:39:00,637 Node[0] Epoch[0] Batch [0-20]   Speed: 692.38 samples/sec       accuracy=0.002232
 336 2020-05-30 22:41:00,435 Node[0] Epoch[0] Batch [20-40]  Speed: 683.82 samples/sec       accuracy=0.001563
 337 2020-05-30 22:43:00,359 Node[0] Epoch[0] Batch [40-60]  Speed: 683.10 samples/sec       accuracy=0.001563
 338 2020-05-30 22:45:01,686 Node[0] Epoch[0] Batch [60-80]  Speed: 675.20 samples/sec       accuracy=0.003125
 339 2020-05-30 22:47:02,563 Node[0] Epoch[0] Batch [80-100] Speed: 677.72 samples/sec       accuracy=0.002344
 340 2020-05-30 22:49:03,353 Node[0] Epoch[0] Batch [100-120]        Speed: 678.21 samples/sec       accuracy=0.005078
 341 2020-05-30 22:51:03,683 Node[0] Epoch[0] Batch [120-140]        Speed: 680.80 samples/sec       accuracy=0.009375
 342 2020-05-30 22:53:03,942 Node[0] Epoch[0] Batch [140-160]        Speed: 681.20 samples/sec       accuracy=0.014844
 343 2020-05-30 22:55:04,840 Node[0] Epoch[0] Batch [160-180]        Speed: 677.60 samples/sec       accuracy=0.015234
 344 2020-05-30 22:57:04,939 Node[0] Epoch[0] Batch [180-200]        Speed: 682.11 samples/sec       accuracy=0.022266
 345 2020-05-30 22:59:05,167 Node[0] Epoch[0] Batch [200-220]        Speed: 681.38 samples/sec       accuracy=0.026953
 346 2020-05-30 23:01:04,916 Node[0] Epoch[0] Batch [220-240]        Speed: 684.10 samples/sec       accuracy=0.035156
 347 2020-05-30 23:03:05,424 Node[0] Epoch[0] Batch [240-260]        Speed: 679.79 samples/sec       accuracy=0.038281
 348 2020-05-30 23:05:05,952 Node[0] Epoch[0] Batch [260-280]        Speed: 679.68 samples/sec       accuracy=0.047266
 349 2020-05-30 23:07:05,842 Node[0] Epoch[0] Batch [280-300]        Speed: 683.30 samples/sec       accuracy=0.050391
2020-05-30 23:08:20,537 Node[12] Epoch[0] Time cost=1884.356                                        
 505 2020-05-30 23:10:24,515 Node[0] Epoch[1] Batch [0-20]   Speed: 681.69 samples/sec       accuracy=0.056920
 506 2020-05-30 23:12:24,895 Node[0] Epoch[1] Batch [20-40]  Speed: 680.51 samples/sec       accuracy=0.061719
 507 2020-05-30 23:14:25,994 Node[0] Epoch[1] Batch [40-60]  Speed: 676.48 samples/sec       accuracy=0.069531
 508 2020-05-30 23:16:26,602 Node[0] Epoch[1] Batch [60-80]  Speed: 679.23 samples/sec       accuracy=0.075391
 509 2020-05-30 23:18:27,119 Node[0] Epoch[1] Batch [80-100] Speed: 679.74 samples/sec       accuracy=0.079297
 510 2020-05-30 23:20:27,847 Node[0] Epoch[1] Batch [100-120]        Speed: 678.56 samples/sec       accuracy=0.100391
 511 2020-05-30 23:22:28,588 Node[0] Epoch[1] Batch [120-140]        Speed: 678.48 samples/sec       accuracy=0.096484
 512 2020-05-30 23:24:29,155 Node[0] Epoch[1] Batch [140-160]        Speed: 679.46 samples/sec       accuracy=0.112891
 513 2020-05-30 23:26:30,442 Node[0] Epoch[1] Batch [160-180]        Speed: 675.43 samples/sec       accuracy=0.114062
 514 2020-05-30 23:28:31,685 Node[0] Epoch[1] Batch [180-200]        Speed: 675.67 samples/sec       accuracy=0.129297
 515 2020-05-30 23:30:33,552 Node[0] Epoch[1] Batch [200-220]        Speed: 672.22 samples/sec       accuracy=0.128906
 516 2020-05-30 23:32:35,014 Node[0] Epoch[1] Batch [220-240]        Speed: 674.45 samples/sec       accuracy=0.132031
 517 2020-05-30 23:34:34,968 Node[0] Epoch[1] Batch [240-260]        Speed: 682.93 samples/sec       accuracy=0.138672
 518 2020-05-30 23:36:35,390 Node[0] Epoch[1] Batch [260-280]        Speed: 680.28 samples/sec       accuracy=0.150781
 519 2020-05-30 23:38:35,667 Node[0] Epoch[1] Batch [280-300]        Speed: 681.10 samples/sec       accuracy=0.158594
2020-05-30 23:41:56,102 Node[0] Epoch[2] Batch [0-20]   Speed: 675.78 samples/sec       accuracy=0.174851
 552 2020-05-30 23:43:59,240 Node[0] Epoch[2] Batch [20-40]  Speed: 665.27 samples/sec       accuracy=0.167578
 553 2020-05-30 23:46:00,746 Node[0] Epoch[2] Batch [40-60]  Speed: 674.21 samples/sec       accuracy=0.185156
 554 2020-05-30 23:48:01,915 Node[0] Epoch[2] Batch [60-80]  Speed: 676.08 samples/sec       accuracy=0.187891
 555 2020-05-30 23:50:02,384 Node[0] Epoch[2] Batch [80-100] Speed: 680.01 samples/sec       accuracy=0.205078
 556 2020-05-30 23:52:03,253 Node[0] Epoch[2] Batch [100-120]        Speed: 677.76 samples/sec       accuracy=0.201953
 557 2020-05-30 23:54:03,607 Node[0] Epoch[2] Batch [120-140]        Speed: 680.66 samples/sec       accuracy=0.221875
 558 2020-05-30 23:56:05,667 Node[0] Epoch[2] Batch [140-160]        Speed: 671.15 samples/sec       accuracy=0.214844
 559 2020-05-30 23:58:06,935 Node[0] Epoch[2] Batch [160-180]        Speed: 675.53 samples/sec       accuracy=0.233203
 560 2020-05-31 00:00:06,959 Node[0] Epoch[2] Batch [180-200]        Speed: 682.53 samples/sec       accuracy=0.248828
 561 2020-05-31 00:02:07,271 Node[0] Epoch[2] Batch [200-220]        Speed: 680.90 samples/sec       accuracy=0.248047
 562 2020-05-31 00:04:08,087 Node[0] Epoch[2] Batch [220-240]        Speed: 678.06 samples/sec       accuracy=0.246094
 563 2020-05-31 00:06:08,070 Node[0] Epoch[2] Batch [240-260]        Speed: 682.76 samples/sec       accuracy=0.248047
 564 2020-05-31 00:08:08,460 Node[0] Epoch[2] Batch [260-280]        Speed: 680.46 samples/sec       accuracy=0.253516
 565 2020-05-31 00:10:09,198 Node[0] Epoch[2] Batch [280-300]        Speed: 678.50 samples/sec       accuracy=0.267188
```

**Comment by xinyu-intel** [comment]
@zhreshold @hetong007

**Comment by zhreshold** [comment]
Any chance it's related to hyperparameter setting or distributed?

**Comment by wuxun-zhang** [comment]
We used the same hyper-parameters for these two tests. We also found that the training accuracy will "jump" between the adjacent epochs when using Gluon Trainer. Not sure if it's related the different update methods of metric in Gluon and symbolic.

**Comment by zhreshold** [comment]
@wuxun-zhang IT's very likely that the metric is smoothing in symbolic training without explicit reset.
